msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase850457.java

```diff
@@ -0,0 +1,134 @@
+import com.microsoft.identity.client.ui.automation.TokenRequestTimeout;
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadPromptHandler;
+import com.microsoft.identity.internal.testutils.labutils.LabConfig;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserHelper;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+
+import org.junit.Test;
+
+// [MSAL] password reset for MAM_CA account
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1177440
+public class TestCase850457 extends AbstractMsalBrokerTest{
+
+    @Test
+    public void test_850457() throws Throwable {
+        final String username = mLoginHint;
+        final String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+        final MsalSdk msalSdk = new MsalSdk();
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .resource(mScopes[0])
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        final MsalAuthResult authResult1 = msalSdk.acquireTokenInteractive(authTestParams, new com.microsoft.identity.client.ui.automation.interaction.OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                        .prompt(PromptParameter.SELECT_ACCOUNT)
+                        .loginHint(mLoginHint)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .speedBumpExpected(false)
+                        .broker(mBroker)
+                        .expectingBrokerAccountChooserActivity(false)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+        authResult1.assertSuccess();
+
+        LabUserHelper.resetPassword(username);
+        TestContext.getTestContext().getTestDevice().getSettings().forwardDeviceTimeForOneDay();
+
+        final MsalAuthResult authResult2 = msalSdk.acquireTokenInteractive(authTestParams, new com.microsoft.identity.client.ui.automation.interaction.OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                        .prompt(PromptParameter.WHEN_REQUIRED)
```

Comment:
I believe we MUST be prompted for password here....is that happening?

Reply:
I am actually thinking adding a new method to PromptHandlerParameters builder to specify something like `expectingPasswordPrompt(true/false)` - I believe this would make the tests more readable. Any thoughts?

Reply:
I should've done that in the first place....not sure why I didn't do that 🤔

Reply:
Yes we are, because there is a check for !parameters.isSessionExpected() which would cause handlePasswordField() to be invoked, but in terms of readability it would help to have expectingPasswordPrompt(bool) so I've added it.

Reply:
Excellent! Thanks a lot...I believe to add that you probably had to also make a change in `uiautomationutilities` in common repo...is there a PR for that?